src/WebJobs.Extensions.DurableTask/LinuxAppServiceILogger.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal abstract class LinuxAppServiceILogger : ILogger
+    {
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            // This logger does not accept adding external scopes,
+            // so it returns a No-Op Disposable, does nothing when disposed.
+            return NoOpDisposable.Instance;
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            // TODO: assuming these logs should never be disabled
+            return true;
+        }
+
+        public string GenerateJsonStringToLog<TState>(TState maybeEventData)
+        {
+            string jsonString = "";
+            if (maybeEventData is EventWrittenEventArgs)
+            {
+                var eventData = maybeEventData as EventWrittenEventArgs;
+
+                var values = eventData.Payload;
+                var keys = eventData.PayloadNames;
+                var keyValuePairs = keys.Zip(values, Tuple.Create);
```

Comment:
This is a very clean way of generating the JSON. However, I'm worried about performance since this is high-volume logging code. `Tuple.Create` will result in the creation of a new object in the managed heap which will need to be garbage collected, and since we will call this method tons of times with lots of data, I expect it to generate tons of garbage and that the CLR garbage collector will eat up a lot of CPU trying to keep up.

As an alternative, I suggest we do some simple looping, like...

```csharp
var values = eventData.Payload;
var keys = eventData.PayloadNames;

var json = new JObject();
for (int i = 0; i < values.Count; i++)
{
   json.Add(keys[i], values[i]);
}
```

Reply:
The functional programmer in me dies a little whenever I see a zip replaced with a for-loop 😄 
But yeah, I was also thinking these would end up being regretful abstractions....thanks for letting me know! Will change it

Reply:
That’s interesting that you consider yourself a functional programmer. I’ve been wanting to offer better Durable support for F# developers for a while. We should talk about this after Python GA. 😊

Reply:
I also love the idea of Functional programming and tried to take inspiration from it where I can, but in practice I am too addicted to state to practice it.